plugin/pkg/scheduler/algorithm/predicates/utils.go

```diff
@@ -74,23 +75,41 @@ func GetEquivalencePod(pod *v1.Pod) interface{} {
 	// For now we only consider pods:
 	// 1. OwnerReferences is Controller
 	// 2. with same OwnerReferences
+	// 3. with same PVC claim
 	// to be equivalent
 	if len(pod.OwnerReferences) != 0 {
 		for _, ref := range pod.OwnerReferences {
 			if *ref.Controller {
 				// a pod can only belongs to one controller
 				return &EquivalencePod{
 					ControllerRef: ref,
+					PVCSet:        getPVCSet(pod),
 				}
 			}
 		}
 	}
 	return nil
 }
 
+// getPVCSet returns a set of PVC names of given pod.
+// Namespace is added as prefix to make every name unique.
+func getPVCSet(pod *v1.Pod) sets.String {
+	result := sets.NewString()
+	for _, volume := range pod.Spec.Volumes {
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+		pvcName := volume.PersistentVolumeClaim.ClaimName
+		result.Insert(pod.GetNamespace() + "-" + pvcName)
```

Comment:
Can we use the PVC UID? I understand this is probably equivalent, but using the UID might make it more obvious. Is there a reason to avoid using the UID? If @bsalamat is ok with this, then maybe we can add a comment explaining it.

Reply:
A user could delete a PVC and create a new one with the same name, but different UID.  The PVC contents could be different, so we would not want to use the cached results for the old PVC.

But actually, how do entries in the equivalence cache get removed?

Reply:
I agree that UID will be more reliable to ensure uniqueness.

Reply:
@msau42 Yes when PVC is updated, the volume predicate cache will be invalidated so this is not a problem.

I was just trying to avoid fetching UID it from scheduler cache since we need to change some methods interface a bit. Not sure if it's allowed during code freeze ...

Anyway it's not a big deal, let's update the patch to reliable way.

Reply:
Oh I see what you mean, the Pod object only contains the PVC name, so you have to lookup the PVC object from the informer cache to get the UID

Reply:
Right, so now I've already refactored it to a factory like `NewEquivalencePodGenerator(args.PVCInfo)`, will update the PR soon after test passes.